clients/src/main/java/org/apache/kafka/common/utils/LogContext.java

```diff
@@ -0,0 +1,374 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+
+public class LogContext {
+
+    private final String logPrefix;
+
+    public LogContext(String logPrefix) {
+        this.logPrefix = logPrefix;
+    }
+
+    public LogContext() {
+        this("");
+    }
+
+    public Logger logger(Class<?> clazz) {
+        return new KafkaLogger(clazz, logPrefix);
+    }
+
+    private static class KafkaLogger implements Logger {
+        private final Logger logger;
+        private final String logPrefix;
+
+        public KafkaLogger(Class<?> clazz, String logPrefix) {
+            this.logger = LoggerFactory.getLogger(clazz);
+            this.logPrefix = logPrefix;
+        }
+
+        @Override
+        public String getName() {
+            return logger.getName();
+        }
+
+        @Override
+        public boolean isTraceEnabled() {
+            return logger.isTraceEnabled();
+        }
+
+        @Override
+        public boolean isTraceEnabled(Marker marker) {
+            return logger.isTraceEnabled(marker);
+        }
+
+        @Override
+        public boolean isDebugEnabled() {
+            return logger.isDebugEnabled();
+        }
+
+        @Override
+        public boolean isDebugEnabled(Marker marker) {
+            return logger.isDebugEnabled(marker);
+        }
+
+        @Override
+        public boolean isInfoEnabled() {
+            return logger.isInfoEnabled();
+        }
+
+        @Override
+        public boolean isInfoEnabled(Marker marker) {
+            return logger.isInfoEnabled(marker);
+        }
+
+        @Override
+        public boolean isWarnEnabled() {
+            return logger.isWarnEnabled();
+        }
+
+        @Override
+        public boolean isWarnEnabled(Marker marker) {
+            return logger.isWarnEnabled(marker);
+        }
+
+        @Override
+        public boolean isErrorEnabled() {
+            return logger.isErrorEnabled();
+        }
+
+        @Override
+        public boolean isErrorEnabled(Marker marker) {
+            return logger.isErrorEnabled(marker);
+        }
+
+        @Override
+        public void trace(String message) {
+            if (logger.isTraceEnabled())
+                logger.trace(logPrefix + message);
+        }
+
+        @Override
+        public void trace(String message, Object arg) {
+            if (logger.isTraceEnabled())
+                logger.trace(logPrefix + message, arg);
+        }
+
+        @Override
+        public void trace(String message, Object arg1, Object arg2) {
+            if (logger.isTraceEnabled())
+                logger.trace(logPrefix + message, arg1, arg2);
```

Comment:
I agree, but I couldn't think of a good alternative. This is also the pattern used in `kafka.utils.Logging` for what it's worth.